### ISSUE TITLE
cannot install in Ubuntu 16

### ISSUE THREAD

**Issue by cdgtf** [robbisg/4dfp_tools#1]
Hi, robbisg,

Thanks for your attention and share! But I find that I cannot install this toolbox in my computer. 
My computer information are as follows: 
OS: Ubuntu 16-10 amd-64 bit
gcc (Ubuntu 6.2.0-5ubuntu12) 6.2.0 20161005
GNU Fortran (Ubuntu 6.2.0-5ubuntu12) 6.2.0 20161005

After I set environmental variables NILSRC and RELEASE directory, I run make_nil-tools.csh file. I cannot successfully compile this toolbox. The error information is as follow:

```
 NILSRC=/home/gtf/4dfp_tools
RELEASE=/home/gtf/4dfp_tools/RELEASE
pushd librms
~/4dfp_tools/librms ~/4dfp_tools 
make -f librms.mak
f77 -O -I4 -e -c deconv.f
f951: Warning: Nonexistent include directory ‘4’ [-Wmissing-include-dirs]
deconv.f:50:14:

       pointer (pa,a)
              1
Error: Cray pointer declaration at (1) requires -fcray-pointer flag
deconv.f:91:16:

       call free(pa)
                1
Error: ‘ptr’ argument of ‘free’ intrinsic at (1) must be INTEGER
librms.mak:37: recipe for target 'deconv.o' failed
make: *** [deconv.o] Error 1
popd
~/4dfp_tools 
pushd imgblur_4dfp
~/4dfp_tools/imgblur_4dfp ~/4dfp_tools 
gcc -O -ffixed-line-length-132 -fcray-pointer -c fimgblur.f
popd
~/4dfp_tools 
pushd img_hist_4dfp
~/4dfp_tools/img_hist_4dfp ~/4dfp_tools 
cc -O -c fimg_mode.c
popd
~/4dfp_tools 
pushd t4imgs_4dfp
~/4dfp_tools/t4imgs_4dfp ~/4dfp_tools 
cc -O -c cvrtflip.c
popd
~/4dfp_tools 
pushd flip_4dfp
~/4dfp_tools/flip_4dfp ~/4dfp_tools 
cc -O -c cflip.c -I/home/gtf/4dfp_tools/TRX
popd
~/4dfp_tools 
pushd parzen_4dfp
~/4dfp_tools/parzen_4dfp ~/4dfp_tools 
gcc -O -ffixed-line-length-132 -fcray-pointer -c imgpac.f
gcc -O -ffixed-line-length-132 -fcray-pointer -c mskpac.f
gcc -O -ffixed-line-length-132 -fcray-pointer -c parzen.f
popd
~/4dfp_tools 
pushd t4_actmapf_4dfp
~/4dfp_tools/t4_actmapf_4dfp ~/4dfp_tools 
gcc -O -ffixed-line-length-132 -fcray-pointer -c ft4ixyz.f
popd
~/4dfp_tools 
pushd imglin
~/4dfp_tools/imglin ~/4dfp_tools 
cc -O -c t4scale.c
cc -O -c dnormal.c
dnormal.c: In function ‘dnormal_test’:
dnormal.c:33:26: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘long int’ [-Wformat=]
  printf ("initilization %d\n", iseed); srand48 (iseed);
                          ^
gcc -O -ffixed-line-length-132 -fcray-pointer -c t4_sub.f
make -f t4_inv.mak release
f77 -O -I4 -e -c t4inv.f
f951: Warning: Nonexistent include directory ‘4’ [-Wmissing-include-dirs]
/usr/bin/ld: warning: cannot find entry symbol -c; defaulting to 00000000000005a0
/usr/lib/gcc/x86_64-linux-gnu/6/../../../x86_64-linux-gnu/Scrt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
/tmp/ccbxxGlj.o: In function `t4inv_':
t4inv.f:(.text+0x81): undefined reference to `geninv_'
collect2: error: ld returned 1 exit status
t4_inv.mak:46: recipe for target 't4inv.o' failed
make: *** [t4inv.o] Error 1
if ( 2 ) exit 2
exit 2
```

Could you please help me check which part I am wrong?

Thanks & Best wishes!
gtf

**Comment by robbisg** [comment]
I'm not sure about this issue, because I made only few changes on the code to make it compile and run.
I had the same Ubuntu, but an older version of the compiler that maybe affect the compiling process thus I made some little changes to cope with my version.
By the way in my docker I had the following instruction:

```
git clone https://github.com/robbisg/4dfp_tools.git $NILSRC
sudo groupadd program
cd $NILSRC && tcsh $NILSRC/make_nil-tools.csh
```

Try it and let me know!

Bests,
R
